components/card/card.js

```diff
@@ -0,0 +1,337 @@
+				z-index: 3;
+			}
+			:host([dir="rtl"]) .d2l-card-actions {
+				left: 0.6rem;
+				right: auto;
+			}
+			.d2l-card-actions ::slotted(*) {
+				margin-left: 0.3rem;
+			}
+			:host([dir="rtl"]) .d2l-card-actions ::slotted(*) {
+				margin-left: 0;
+				margin-right: 0.3rem;
+			}
+			.d2l-card-badge {
+				line-height: 0;
+				padding: 0 0.8rem;
+			}
+			.d2l-card-footer {
+				box-sizing: border-box;
+				flex: none;
+				padding: 1.2rem 0.8rem 0.6rem 0.8rem;
+				pointer-events: none;
+				width: 100%;
+				z-index: 2;
+			}
+			:host([align-center]) .d2l-card-footer {
+				text-align: center;
+			}
+
+			.d2l-card-footer ::slotted([slot="footer"]) {
+				pointer-events: all;
+			}
+
+			.d2l-card-footer-hidden .d2l-card-footer {
+				box-sizing: content-box;
+				height: auto;
+			}
+
+			:host([subtle]) {
+				border: none;
+				box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.03);
+			}
+			:host(:hover),
+			:host([subtle]:hover) {
+				transform: translateY(-4px);
+			}
+			:host(:hover) {
+				box-shadow: 0 2px 14px 1px rgba(0, 0, 0, 0.06);
+			}
+			:host([subtle]:hover) {
+				box-shadow: 0 4px 18px 2px rgba(0, 0, 0, 0.06);
+			}
+			:host([_active]) {
+				border-color: transparent;
+				box-shadow: 0 0 0 2px #ffffff, 0 0 0 4px var(--d2l-color-celestine);
+			}
+			:host([_active]:hover),
+			:host([subtle][_active]:hover) {
+				border-color: transparent;
+				box-shadow: 0 0 0 2px #ffffff, 0 0 0 4px var(--d2l-color-celestine);
+				transform: translateY(-4px);
+			}
+			/* .d2l-card-link-container-hover is used to only color/underline when
+			hovering the anchor; these styles are not applied when hovering actions */
+			:host([href]) .d2l-card-link-container-hover,
+			:host([href][_active]) .d2l-card-content {
+				color: var(--d2l-color-celestine);
+				text-decoration: underline;
+			}
+			/* this is needed to ensure tooltip is not be clipped by adjacent cards */
+			:host([_tooltip_showing]) {
+				z-index: 1;
+			}
+			/* this is needed to ensure open menu will be ontop of adjacent cards */
+			:host([_dropdown-action-open]) {
+				z-index: 2;
+			}
+			:host(:not([href])),
+			:host([subtle]:not([href])) {
+				box-shadow: none;
+				transform: none;
+			}
+			@media (prefers-reduced-motion: reduce) {
+				:host {
+					transition: none;
+				}
+			}
+		`];
+	}
+
+	constructor() {
+		super();
+		this.alignCenter = false;
+		this.download = false;
+		this.subtle = false;
+		this._active = false;
+		this._dropdownActionOpen = false;
+		this._footerHidden = true;
+		this._hover = false;
+		this._tooltipShowing = false;
+		this._onBadgeResize = this._onBadgeResize.bind(this);
+		this._onFooterResize = this._onFooterResize.bind(this);
+	}
+
+	firstUpdated(changedProperties) {
+		super.firstUpdated(changedProperties);
+		const badgeObserver = new ResizeObserver(this._onBadgeResize);
+		badgeObserver.observe(this.shadowRoot.querySelector('.d2l-card-badge'));
```

Comment:
Do we need to worry about unobserving these?

Reply:
Good question.  I don't think so, since the observer is scoped to this component and will get cleaned up along with attached handlers when the component is garbage collected.